pootle/apps/import_export/utils.py

```diff
@@ -62,3 +72,68 @@ def import_file(f, user=None):
         # This should not happen!
         logger.error("Error importing file: %s", str(e))
         raise FileImportError(_("There was an error uploading your file"))
+
+
+class TPTMXExporter(object):
+
+    def __init__(self, context):
+        self.context = context
+
+    @property
+    def exported_revision(self):
+        return revision.get(self.context.__class__)(
+            self.context).get(key="pootle.offline.tm")
+
+    @cached_property
+    def revision(self):
+        return revision.get(self.context.__class__)(
+            self.context.directory).get(key="stats")
+
+    def update_exported_revision(self):
+        if self.has_changes():
+            revision.get(self.context.__class__)(
+                self.context).set(keys=["pootle.offline.tm"],
+                                  value=self.revision)
+
+    def has_changes(self):
+        return self.revision != self.exported_revision
+
+    @property
+    def directory(self):
+        return os.path.join(settings.MEDIA_ROOT,
+                            'offline_tm',
+                            self.context.language.code)
+
+    @property
+    def filename(self):
+        filename = self.context.project.fullname.replace(' ', '_')
+        filename = ".".join([filename, self.revision, 'tmx'])
+        return "%s.zip" % filename
```

Comment:
No target language in the output filename? How would you differentiate two export files for the same project but different languages?